custom_logs.tf

```diff
@@ -0,0 +1,177 @@
+locals {
+  columns = {
+    "state"                  = "string"
+    "TimeGenerated"          = "datetime"
+    "additional_information" = "string"
+    "affected_entity"        = "string"
+    "affected_object"        = "string"
+    "monitor_description"    = "string"
+    "monitor_name"           = "string"
+    "monitor_package"        = "string"
+    "script_name"            = "string"
+    "script_version"         = "string"
+    "threshold"              = "string"
+    "value"                  = "string"
+  }
+  kql_types = join(", ", [for column_name, column_type in local.columns : "${column_name}:${column_type}"])
+
+  resource_group_id = substr(var.log_analytics_workspace.id, 0, length(regexall("^[^/]*/[^/]*/[^/]*/[^/]*/[^/]*", var.log_analytics_workspace.id)[0]))
+}
+
+resource "azapi_resource" "data_collection_json_logs_table" {
+  name      = "MonitoringScriptsJSON_CL"
```

Comment:
In my opinion, this one should be MonitoringScripts_CL and the other one MonitoringScriptsTEXT_CL, since hopefully we will keep this one and eventually remove the TEXT one. Alternatively, name both JSON and TEXT respectively for uniformity.

Reply:
renamed the the TEXT logs for uniformity